web/app/view/MapMarkerController.js

```diff
@@ -488,5 +494,56 @@ Ext.define('Traccar.view.MapMarkerController', {
         } else if (point) {
             this.getView().getMapView().fit(new ol.geom.Point(point));
         }
+    },
+
+    updateVisibleDevice: function (store, data) {
+        var i, device;
+
+        if (!Ext.isArray(data)) {
+            data = [data];
+        }
+
+        for (i = 0; i < data.length; i++) {
+            device = data[i];
+            if (device.get('id') in this.latestMarkers) {
+                this.checkDeviceVisibility(device);
+            }
+        }
+    },
+
+    checkDeviceVisibility: function (device) {
+        var deviceId, accuracy, liveLine, marker;
+        deviceId = device.get('id');
+        marker = this.latestMarkers[deviceId];
+        accuracy = this.accuracyCircles[deviceId];
+        liveLine = this.liveRoutes[deviceId];
+        if (Ext.getStore('VisibleDevices').contains(device)) {
+            if (marker && !this.getView().getMarkersSource().getFeatureById(marker.getId())) {
```

Comment:
It throws this exception https://openlayers.org/en/v4.0.0/doc/errors/#30 if feature already there
and this then try to remove not existed feature
![image](https://cloud.githubusercontent.com/assets/5688080/24232304/d8b18aa6-0fab-11e7-9cbd-f40ea0a51513.png)